File: Python3 Solutions/190. Reverse Bits.py
# Question Link: https://leetcode.com/problems/reverse-bits/

class Solution:
    def reverseBits(self, n: int) -> int:
        left_pointer = 2**31
        reverse = 0
        
        while n > 0:
            reverse += left_pointer * (n & 1)
            left_pointer >>= 1
            n >>= 1
        
        return reverse

File: Camp Progress/190. Reverse Bits.cpp
// Question Link: https://leetcode.com/problems/reverse-bits/

class Solution {
public:
    uint32_t reverseBits(uint32_t n) {
        uint32_t leftPointer = pow(2, 31); // -1 is for 0
        uint32_t reversed = 0;
        while (n > 0) {
            reversed += leftPointer * (n & 1);
            leftPointer >>= 1;
            n >>= 1;
        }
        
        return reversed;
    }
};